runtime/src/in_mem_accounts_index.rs

```diff
@@ -235,34 +235,96 @@ impl<T: IndexValue> InMemAccountsIndex<T> {
     ) -> RT {
         self.get_only_in_mem(pubkey, |entry| {
             if let Some(entry) = entry {
+                if entry.disk_unknown() {
+                    // disk_unknown is marked, load from the disk and merge with in-mem entry
+                    // we found the pubkey
+                    // but, we haven't checked to see if there is more on disk
+                    // so, we have to load from disk first, then merge with in-mem
+                    // right now we have a read lock on the in-mem idx, fwiw
+                    let disk_entry = self.load_account_entry_from_disk(pubkey);
```

Comment:
should consider grabbing an arc clone of the entry and then exiting the scope to get rid of the read lock on the bin.

Reply:
Can you elaborate more on this?

Reply:
```
let mut entry_outside_lock = None;
self.get_only_in_mem(...) {
...
if entry.disk_unkonwn() {
entry_outside_lock = Some(Arc::clone(&entry)); // this grabs a hardcount to the entry in  mem
return; // this drops bin's read lock for the entire bin
}
...
}

// this runs outside the bin's read lock
if let Some(entry_outside_lock) = entry_outside_lock {
let disk_entry = self.load_account_entry_from_disk(pubkey);
do merge here
}
```

I think you can wait to do this. Just thinking that we want to avoid holding locks while we have page faults to read things from disk.